stripe/src/main/java/com/stripe/android/paymentsheet/PaymentSheetActivity.kt

```diff
@@ -91,6 +95,23 @@ internal class PaymentSheetActivity : BasePaymentSheetActivity<PaymentResult>()
         PaymentConfiguration.getInstance(application)
     }
 
+    private val buyButtonLabel: String
+        get() = resources.getString(
+            R.string.stripe_paymentsheet_pay_button_amount
+        )
+
+    private val currencyFormatter = CurrencyFormatter()
+    private fun getLabelText(viewState: ViewState.Buy.Ready) : String {
+        val currency = Currency.getInstance(
+            viewState.currencyCode.toUpperCase(Locale.ROOT))
+
+        return String.format(
+            "%s %s",
+            buyButtonLabel,
+            currencyFormatter.format(viewState.amount, currency)
+        )
```

Comment:
Use Android string resources for interpolation - they will apply the correct Locale.

Reply:
I am not sure what you mean. Is there a locale formatting for something here?  The label already has the locale, and the currentFormatter doens't have anything locale specific

Reply:
You're right, I think my comment only told half the story. It will also properly support localized text. For example, in other languages the string "Pay $10.00" may not have the parts of the sentence in that order. There's usually not a good reason for using `String.format()` for any text that will be rendered on the screen.